rpc/src/commonMain/kotlin/foundation/metaplex/rpc/RpcInterface.kt

```diff
@@ -0,0 +1,84 @@
+package foundation.metaplex.rpc
+
+import foundation.metaplex.solanapublickeys.PublicKey
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+
+/**
+ * Represents an interface for making remote procedure calls (RPC).
+ */
+interface RpcInterface {
+
+    /**
+     * Retrieves account information for a given public key.
+     *
+     * @param publicKey The public key of the account.
+     * @param configuration Optional configuration for the RPC request. Defaults to Base64 Encoding
+     * @param serializer The serializer used for deserializing the response.
+     * @return An [Account] object representing the account information, or null if not found.
+     */
+    suspend fun <T> getAccountInfo(
+        publicKey: PublicKey,
+        configuration: RpcGetAccountInfoConfiguration?,
+        serializer: KSerializer<T>,
+    ): Account<T>?
+}
+
+/**
+ * Configuration input for the [RpcInterface.getAccountInfo] function.
+ */
+@Serializable
+data class RpcGetAccountInfoConfiguration(
+    val encoding: Encoding = Encoding.base64,
+    val commitment: Commitment? = null,
+    val minContextSlot: UInt? = null,
+    val dataSlice: RpcDataSlice? = null
+)
+
+/**
+ * Enumeration representing the commitment level for RPC requests.
+ */
+@Serializable
+enum class Commitment {
```

Comment:
I might advocate for putting all of these data classes in a file alongside the interface. Just to keep definitions clear. I can be talked out of this though.

Reply:
Arent they in the same file?

Reply:
Yeah I just mean all those little helper classes should be in a _different_ file from the RpcInterface definition. Just for cleanliness.